src/PathContainer.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace Dhii\Container;
+
+use Psr\Container\ContainerInterface;
+use function array_filter;
+use function ltrim;
+
+/**
+ * A container implementation that wraps around another to provide access to its values via partial path-like keys.
+ *
+ * @since [*next-version*]
+ */
+class PathContainer implements ContainerInterface
```

Comment:
What are the practical benefits, i.e. realistic case scenarios of using this? I understand that this is supposed to work on inner containers which have paths for keys. In effect, doesn't this actually work as a prefixing container?

Reply:
Also, judging by the name of the class, I kind of expected it to instead retrieve values from a hierarchy of containers, i.e.

```
[
	'my' => [
		'path' => 'value',
	],
]
```

Running `PathContainer->get('my/path')` would return `'value'` for the above container hierarchy. I would like to have such an implementation. Are you planning to add this? And if not, when I add it, not sure what to name it then :D

Reply:
Yeah, needs a better name.

If a container has the key `my/special/path`, the `PathContainer` decorator would allow you to get, for instance, `my/special` (or any other non-existent key for that matter). In those cases, it creates a clone of itself that knows about that key, so when you call `path`, you finally get the value.

So you're right. It kinda does sound like a recursive prefixing container. Until an actual key is found, it will keep creating containers that are prefixed. Now I'm thinking if the concept can be generalized to some kind of generic recursive container decorator, which coupled with the prefixing decorator would do the same thing as this class.

In any case, how about `RecursivePrefixingContainer`?

Reply:
In my understanding, what this class does is kinda create a segment of the inner container's values, without actually retrieving that segment. And I can definitely see the use for this, given that there may be a value at `my/deep`, and the value at `my/deep/key`, where `key` is not "inside" `my/deep`. This class would circumvent that by still returning results that are consistent with configuration.

So, how about a `SegmentingContainer`?

Reply:
I like it.